File: backend/app/models/provider.py
"""
模型提供方配置模型
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict
from datetime import datetime


class ModelProvider(BaseModel):
    provider_id: str
    name: str
    provider_type: str
    base_url: Optional[str] = None
    api_key: Optional[str] = None
    api_version: Optional[str] = None
    model: str
    extra_headers: Dict[str, str] = Field(default_factory=dict)
    is_active: bool = False
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)

File: frontend/src/types/provider.ts
// 模型提供方类型定义

export type ProviderType =
  | 'openai'
  | 'azure'
  | 'anthropic'
  | 'deepseek'
  | 'qwen'
  | 'ollama'
  | 'custom';

export interface ModelProvider {
  provider_id: string;
  name: string;
  provider_type: ProviderType;
  base_url?: string | null;
  api_key?: string | null;
  api_version?: string | null;
  model: string;
  extra_headers?: Record<string, string>;
  is_active: boolean;
  created_at?: string;
}
